### PR TITLE
Design Guide Improvement

### PR DESCRIPTION
**Pull request by StevenGann**
Cleaned up the design guide a little to clarify a some of the language, be a bit more brand neutral, and list some alternative EEPROMs that are compatible with the specifications.

### CODE REVIEW

**Comment by JamesH65** [comment]
@jimbojr Are you the HAT docs go to guy?

File: designguide.md [hunk 1]
```diff
@@ -54,11 +54,22 @@ Pull-ups must be provided on the top board for ID_SC and ID_SD (SCL and SDA resp
 - Devices that perform I2C clock stretching are not supported.
 - Write protect pin must be supported and protect the entire device memory.
 
-Note that due to the restrictions above (only using non-paged 16-bit addressable devices is allowed), many of the smaller I2C EEPROMs are ruled out - please check datasheets carefully when choosing a suitable EEPROM for your HAT.
+**Note:** Due to the restrictions above (only using non-paged 16-bit addressable devices is allowed), many of the smaller and larger I2C EEPROMs are ruled out - please check datasheets carefully when choosing a suitable EEPROM for your HAT.
 
-A recommended part that satisfies the above constraints is OnSemi CAT24C32 which is a 32kbit (4kbyte) device. The minimum EEPROM size required is variable and depends on the size of the vendor data strings in the EEPROM and whether a device tree data blob is included (and its size) and whether any other vendor specific data is included.
+The minimum EEPROM size required is variable and depends on the size of the vendor data strings in the EEPROM and whether a device tree data blob is included (and its size) and whether any other vendor specific data is included.
```
Comment (pelwell) [inline designguide.md:59]: The and after EEPROM should be replaced by a comma,

**Review by pelwell — COMMENTED** [review]
I'm okay with the changes in general, although see minor edit, but I think it's disingenuous of you not to declare what I think is an affiliation to Atmel.

**Comment by StevenGann** [comment]
I am a Microchip employee and it is clearly declared on my profile, but I don't think I need to wear a special hat or blow a horn because of it whenever I do something in my free time. I've clearly stated my affiliations and it's plainly visible that my edits are brand-neutral and improve representation while the original was highly biased and made false claims to promote that bias.

I stand by my edits unless someone can explain why only Microchip EEPROMs are warned against when every manufacturer, including the officially endorsed OnSemi, offers the same partial-array protection feature in some variants.

**Comment by JamesH65** [comment]
On the whole, the documentation aim is to REDUCE the amount of recommendations we make as these always come back to haunt at some point.  To that end I'd suggest removing any reference to third party products at all. i.e. remote the OnSemi recommendation, and the table. That way we are not targeted with claims of favouritism from any direction.

**Comment by rgetz** [comment]
> On the whole, the documentation aim is to REDUCE the amount of recommendations we make as these always come back to haunt at some point.

As a developer - I like it when validated/verified part numbers are shared in the doc. It indicates how what (if any) has been tested and validated. A list of what is *known* to work is better than a spec...

**Comment by JamesH65** [comment]
But as you can see, a third party wants to add their chip, which WE haven't tested. So what should we do? We don't do validation as a matter of course on stuff we don't use.

**Comment by rgetz** [comment]
That's easy - don't add it.

Or clearly state that during your validation/verification/release process - the ones noted (*) are tested, and the others aren't, but have been reported to work by others.

That way developers know there are alternatives, but you don't get stuck answering questions on hardware you don't have.

File: designguide.md [hunk 2]
```diff
@@ -54,11 +54,22 @@ Pull-ups must be provided on the top board for ID_SC and ID_SD (SCL and SDA resp
...
+Manufacturer | Density | Part Number
+------------ | ------- | -----------
+Atmel        | 32Kbit  | AT24C32E
+Atmel        | 256Kbit | AT24C256C
+OnSemi       | 32Kbit  | CAT24C32
+OnSemi       | 256Kbit | CAT24C256
+ST Micro     | 32Kbit  | M24C32-F
+ST Micro     | 256Kbit | M24256-BR
+
+It is recommended that EEPROM WP (write protect) pin be connected to a test point on the board and pulled up to 3V3 with a 1K resistor. The idea is that at board test/probe the EEPROM can be written (WP pin can be driven LOW), but there is no danger of a user accidentally changing the device contents once the board leaves the factory. Note that on some devices WP does not write protect the entire array (often called half-array, quarter-array, or partial-array write protection) – avoid using these.
```
Comment (pelwell) [inline designguide.md:72]: The deletion of the line "Note that the recommended device has an internal pull down hence the stiff (1K) pull up is required" is a regression. Reword it to fit your removal of a specific recommended device, but the hint should remain.

But for this I would have merged it now.
